rasa/core/training/interactive.py

```diff
@@ -748,6 +748,9 @@ def _collect_messages(events: List[Dict[Text, Any]]) -> List[Message]:
             msg = Message.build(data["text"], data["intent"]["name"], data["entities"])
             msgs.append(msg)
 
+        elif event.get("event") == UserUtteranceReverted.type_name:
+            msgs.pop()  # user corrected the nlu, remove incorrect example
```

Comment:
Do you know how this interacts with `action_back` which is two UURs in a row?

Reply:
It removes the data from the latest user message and the message that triggered `action_back`.  Seems like intended behaviour to me - we give the users the option to fork anyway